Startup.cs

```diff
@@ -63,10 +64,19 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllersWithViews();
             services.AddRazorPages();
-            services.AddAutoMapper(typeof(CommentMappingProfile), typeof(ThreadMappingProfile));
+
+            var commentRenderer = new CommentRenderService();
+            services.AddSingleton<CommentRenderService>(commentRenderer);
+            services.AddAutoMapperBuilder(builder =>
+            {
+                builder.Profiles.Add(new CommentMappingProfile(commentRenderer));
```

Comment:
Standard DI wasn't working for the mapper for some reason so this is the workaround.